tests/io/test_h5io.py

```diff
@@ -0,0 +1,54 @@
+#
+# Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+#
+# NVIDIA CORPORATION and its licensors retain all intellectual property
+# and proprietary rights in and to this software, related documentation
+# and any modifications thereto.  Any use, reproduction, disclosure or
+# distribution of this software and related documentation without an express
+# license agreement from NVIDIA CORPORATION is strictly prohibited.
+#
+
+from atacworks.io import h5io
+import h5py
+import numpy as np
+import pytest
+import os
+
+
+@pytest.mark.cpu
+def test_dict_to_h5(tmpdir):
+    """Test dict to h5."""
+
+    input_dict = {"features": np.array([0, 1, 2, 3]),
+                  "scores": np.array([1, 2, 1, 1])}
+    h5file = os.path.join(tmpdir, "h5file.h5")
+    h5io.dict_to_h5(input_dict, h5file)
+    with h5py.File(h5file, 'r') as f:
+        for key in input_dict.keys():
+            data = np.array(f[key])
+            assert data.all() == input_dict[key].all()
+
+    # Append to existing dataset and test.
+    input_dict = {"features": np.array([0, 1, 2, 3]),
+                  "scores": np.array([1, 2, 1, 1])}
+    new_dict = {"features": np.array([0, 1, 2, 3, 4, 5, 6])}
+    input_dict.update(new_dict)
```

Comment:
How about this to update the `features entry`
```
input_dict["features"] = np.array([0, 1, 2, 3, 4, 5, 6])
```

Reply:
You think it's better to manually update the input_dict, instead of updating using new_dict ? It depends on whether we trust the `.update` or not. I think both are fine, if you have a strong preference, I'll change it.

Reply:
I was mostly wondering if there's a reason to use `update` instead of overwriting? Okay either way

Reply:
So, I send in a "new_dict" dictionary to verify that the API is actually appending to the existing dataset instead of overwriting it. So, I update the `input_dict` to reflect that.

Reply:
done!